ContentType Pull does NOT support JSON Field

### Parent Issue

_No response_

### Problem Statement

If I create a CT with JsonField then I use the global Pull to build a workspace I get the following error:

```
[ERROR] ❗  Error in command [pull] with message:
  Could not resolve type id 'com.dotcms.contenttype.model.field.ImmutableJSONField' as a subtype of `com.dotcms.contenttype.model.field.Field`: no such class found
 at [Source: (org.jboss.resteasy.sp...
com.fasterxml.jackson.databind.exc.InvalidTypeIdException: Could not resolve type id 'com.dotcms.contenttype.model.field.ImmutableJSONField' as a subtype of `com.dotcms.contenttype.model.field.Field`: no such class found
 at [Source: (org.jboss.resteasy.specimpl.AbstractBuiltResponse$InputStreamWrapper); line: 1, column: 215163] (through reference chain: com.dotcms.model.ResponseEntityView$Builder["entity"]->java.util.ArrayList[24]->com.dotcms.contenttype.model.type.ImmutableSimpleContentType$Json["layout"]->java.util.ArrayList[0]->com.dotcms.contenttype.model.field.ImmutableFieldLayoutRow$Json["columns"]->java.util.ArrayList[0]->com.dotcms.contenttype.model.field.ImmutableFieldLayoutColumn$Json["fields"]->java.util.ArrayList[4])
```

### Steps to Reproduce

1. create a CT with JsonField 
2. use the global Pull 

### Acceptance Criteria

All CT fields must be supported 

### dotCMS Version

master

### Proposed Objective

Customer Success

### Proposed Priority

Please Select

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/provider/ClientObjectMapper.java
+++ b/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/provider/ClientObjectMapper.java
@@ -13,6 +13,7 @@ import com.dotcms.contenttype.model.field.ImmutableFileField;
 import com.dotcms.contenttype.model.field.ImmutableHiddenField;
 import com.dotcms.contenttype.model.field.ImmutableHostFolderField;
 import com.dotcms.contenttype.model.field.ImmutableImageField;
+import com.dotcms.contenttype.model.field.ImmutableJSONField;
 import com.dotcms.contenttype.model.field.ImmutableKeyValueField;
 import com.dotcms.contenttype.model.field.ImmutableLineDividerField;
 import com.dotcms.contenttype.model.field.ImmutableMultiSelectField;
@@ -130,6 +131,7 @@ public class ClientObjectMapper implements ContextResolver<ObjectMapper> {
                 .allowIfSubType(ImmutableTextField.class)
                 .allowIfSubType(ImmutableTimeField.class)
                 .allowIfSubType(ImmutableWysiwygField.class)
+                .allowIfSubType(ImmutableJSONField.class)
 
                 .allowIfSubType(Map.class)
                 .allowIfSubType(List.class)

--- a/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/field/Field.java
+++ b/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/field/Field.java
@@ -57,6 +57,7 @@ import org.immutables.value.Value.Default;
         @Type(value = WysiwygField.class),
         @Type(value = RowField.class),
         @Type(value = ColumnField.class),
+        @Type(value = JSONField.class),
 })
 @JsonInclude(Include.NON_DEFAULT)
 @Value.Style(passAnnotations = {JsonInclude.class})

--- a/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/field/JSONField.java
+++ b/tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/contenttype/model/field/JSONField.java
@@ -1,0 +1,17 @@
+package com.dotcms.contenttype.model.field;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableJSONField.class)
+@JsonDeserialize(as = ImmutableJSONField.class)
+public abstract class JSONField extends Field{
+
+    @Value.Default
+    @Override
+    public DataTypes dataType(){
+        return DataTypes.LONG_TEXT;
+    }
+}

--- a/tools/dotcms-cli/api-data-model/src/test/java/com/dotcms/api/ContentTypeAPIIT.java
+++ b/tools/dotcms-cli/api-data-model/src/test/java/com/dotcms/api/ContentTypeAPIIT.java
@@ -445,7 +445,11 @@ class ContentTypeAPIIT {
                 + "\t}, {\n"
                 + "\t\t\"clazz\": \"ColumnField\",\n"
                 + "\t\t\"variable\": \"col\"\n"
-                + "\t}]\n"
+                + "\t}, {\n"
+                + "\t\t\"clazz\": \"JSONField\",\n"
+                + "\t\t\"variable\": \"json\"\n"
+                + "\t}"
+                + "]\n"
                 + "}";
 
         final ObjectMapper objectMapper = new ClientObjectMapper().getContext(null);

